common/chan_utils.go

```diff
@@ -16,15 +16,18 @@
 
 package common
 
-import "context"
+import (
+	"context"
+)
 
 func SendToChannel[T any](ctx context.Context, c chan<- T, e interface{}) bool {
 	select {
-	case c <- e.(T): // It will panic if `e` is not of type `T` or a type that can be converted to `T`.
-		return true
 	case <-ctx.Done():
 		close(c)
 		return false
+	default:
+		c <- e.(T)
```

Comment:
This change will lose the functionality of the `ctx`.
It may be blocked here and if the ctx is Done, it won't be unblocked.

We also need to add this case to cover it: If the `SendToChannel` is blocked, we could also use ctx to unblock it.